composer.json

```diff
@@ -30,53 +30,53 @@
         "pusher/pusher-php-server": "<2.2.1",
         "sabre/dav": ">=1.6,<1.6.99|>=1.7,<1.7.11|>=1.8,<1.8.9",
-        "shopware/shopware": "<4.3.7|>=5,<5.1|>=5.1,<5.1.5",
+        "shopware/shopware": "<4.3.7|>=5,<5.1.5",
         "silverstripe/cms": ">=3.1,<3.1.11|>=3,<=3.0.11",
         "silverstripe/forum": "<=0.6.1|>=0.7,<=0.7.3",
-        "silverstripe/framework": ">=3,<3.1|>=3.1,<3.2|>=3.2,<3.3",
+        "silverstripe/framework": ">=3,<3.3",
         "silverstripe/userforms": "<3",
         "simplesamlphp/simplesamlphp": "<1.14.4",
         "socalnick/scn-social-auth": "<1.15.2",
         "swiftmailer/swiftmailer": ">=4,<4.99.99|>=5,<5.2.1",
         "symfony/dependency-injection": ">=2,<2.0.17",
-        "symfony/form": ">=2.3,<2.3.35|>=2.4,<2.5|>=2.5,<2.6|>=2.6,<2.6.12|>=2.7,<2.7.7",
-        "symfony/framework-bundle": ">=2,<2.1|>=2.1,<2.2|>=2.2,<2.3|>=2.3,<2.3.18|>=2.4,<2.4.8|>=2.5,<2.5.2",
-        "symfony/http-foundation": ">=2,<2.1|>=2.1,<2.2|>=2.2,<2.3|>=2.3,<2.3.27|>=2.4,<2.5|>=2.5,<2.5.11|>=2.6,<2.6.6",
-        "symfony/http-kernel": ">=2,<2.1|>=2.1,<2.2|>=2.2,<2.3|>=2.3,<2.3.29|>=2.4,<2.5|>=2.5,<2.5.12|>=2.6,<2.6.8",
+        "symfony/form": ">=2.3,<2.3.35|>=2.4,<2.6.12|>=2.7,<2.7.7",
+        "symfony/framework-bundle": ">=2,<2.3.18|>=2.4,<2.4.8|>=2.5,<2.5.2",
+        "symfony/http-foundation": ">=2,<2.3.27|>=2.4,<2.5.11|>=2.6,<2.6.6",
+        "symfony/http-kernel": ">=2,<2.3.29|>=2.4,<2.5.12|>=2.6,<2.6.8",
         "symfony/routing": ">=2,<2.0.19",
-        "symfony/security": ">=2.3,<2.3.37|>=2.4,<2.5|>=2.5,<2.6|>=2.6,<2.6.13|>=2.7,<2.7.9|>=2,<2.0.25|>=2.1,<2.1.13|>=2.2,<2.2.9",
-        "symfony/security-core": ">=2.8,<2.8.6|>=3,<3.0.6|>=2.4,<2.5|>=2.5,<2.6|>=2.6,<2.6.13|>=2.7,<2.7.9",
-        "symfony/security-http": ">=2.4,<2.5|>=2.5,<2.6|>=2.6,<2.7|>=2.7,<2.7.13|>=2.3,<2.3.41|>=2.8,<2.8.6|>=3,<3.0.6",
+        "symfony/security": ">=2,<2.0.25|>=2.1,<2.1.13|>=2.2,<2.2.9|>=2.3,<2.3.37|>=2.4,<2.6.13|>=2.7,<2.7.9",
+        "symfony/security-core": ">=2.4,<2.6.13|>=2.7,<2.7.9|>=2.8,<2.8.6|>=3,<3.0.6",
+        "symfony/security-http": ">=2.4,<2.7.13|>=2.3,<2.3.41|>=2.8,<2.8.6|>=3,<3.0.6",
         "symfony/serializer": ">=2,<2.0.11",
-        "symfony/symfony": ">=2,<2.1|>=2.1,<2.2|>=2.2,<2.3|>=2.3,<2.3.41|>=2.4,<2.5|>=2.5,<2.6|>=2.6,<2.7|>=2.7,<2.7.13|>=2.8,<2.8.6|>=3,<3.0.6",
+        "symfony/symfony": ">=2,<2.3.41|>=2.4,<2.7.13|>=2.8,<2.8.6|>=3,<3.0.6",
         "symfony/translation": ">=2,<2.0.17",
         "symfony/validator": ">=2,<2.0.24|>=2.1,<2.1.12|>=2.2,<2.2.5|>=2.3,<2.3.3",
-        "symfony/web-profiler-bundle": ">=2,<2.1|>=2.1,<2.2|>=2.2,<2.3|>=2.3,<2.3.19|>=2.4,<2.4.9|>=2.5,<2.5.4",
+        "symfony/web-profiler-bundle": ">=2,<2.3.19|>=2.4,<2.4.9|>=2.5,<2.5.4",
         "symfony/yaml": ">=2,<2.0.22|>=2.1,<2.1.7",
         "thelia/backoffice-default-template": ">=2.1,<2.1.2",
-        "thelia/thelia": ">=2.1.0-beta1,<2.1.3|>=2.1,<2.1.2",
+        "thelia/thelia": ">=2.1,<2.1.2|>=2.1.0-beta1,<2.1.3",
```

Comment:
@Ocramius next step would be recognizing such overlapping ranges to merge them properly to ``>= 2.1.0-beta1,<2.1.3``. The logic detecting this seems to be broken for beta versions.